File: apps/hrms-employee/src/main/java/com/cre/hrms/employee/EmployeeApplication.java
package com.cre.hrms.employee;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication(scanBasePackages = {
    "com.cre.hrms.employee",
    "com.cre.hrms.common",
    "com.cre.hrms.core",
    "com.cre.hrms.persistence",
    "com.cre.hrms.security",
    "com.cre.hrms.config"
})
public class EmployeeApplication {
    public static void main(String[] args) {
        SpringApplication.run(EmployeeApplication.class, args);
    }
}

File: apps/hrms-employee/src/main/kotlin/com/cre/hrms/employee/EmployeeApplication.kt
package com.cre.hrms.employee

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.data.jpa.repository.config.EnableJpaRepositories

@SpringBootApplication
@ComponentScan(basePackages = [
    "com.cre.hrms.employee",
    "com.cre.hrms.security",
    "com.cre.hrms.messaging"
])
@EnableJpaRepositories(basePackages = [
    "com.cre.hrms.persistence.employee",
    "com.cre.hrms.persistence.department"
])
@EntityScan(basePackages = [
    "com.cre.hrms.persistence.employee",
    "com.cre.hrms.persistence.department"
])
class EmployeeApplication

fun main(args: Array<String>) {
    runApplication<EmployeeApplication>(*args)
}
